diff --git a/Telegram/SourceFiles/data/components/sponsored_messages.cpp b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
index 04b421ef67..8e5ea6e259 100644
--- a/Telegram/SourceFiles/data/components/sponsored_messages.cpp
+++ b/Telegram/SourceFiles/data/components/sponsored_messages.cpp
@@ -302,6 +302,7 @@ HistoryItem *SponsoredMessages::injectItem(
 }
 
 bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
+	return false;
 	if (history->peer->isChannel()) {
 		return true;
 	} else if (const auto user = history->peer->asUser()) {
@@ -311,6 +312,7 @@ bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
 }
 
 bool SponsoredMessages::canHaveFor(not_null<HistoryItem*> item) const {
+	return false;
 	return item->history()->peer->isBroadcast()
 		&& item->isRegular();
 }
diff --git a/Telegram/SourceFiles/dialogs/dialogs_inner_widget.cpp b/Telegram/SourceFiles/dialogs/dialogs_inner_widget.cpp
index cfcdaf9a50..0bf7532e65 100644
--- a/Telegram/SourceFiles/dialogs/dialogs_inner_widget.cpp
+++ b/Telegram/SourceFiles/dialogs/dialogs_inner_widget.cpp
@@ -4741,6 +4741,7 @@ void InnerWidget::peerSearchReceived(Api::PeerSearchResult result) {
 		return false;
 	};
 	auto added = base::flat_set<not_null<PeerData*>>();
+	if (false)
 	for (const auto &sponsored : result.sponsored) {
 		const auto peer = sponsored.peer;
 		if (inlist(peer) || _sponsoredRemoved.contains(peer)) {
